src/common/docker-compose/index.ts

```diff
@@ -142,7 +127,11 @@ export const generate = (dependency_manager: LocalDependencyManager, build_prod
       service_to.restart = 'always';
       compose.services[node_to.normalized_ref].depends_on.push(node_from.normalized_ref);
     } else if (edge instanceof ServiceEdge) {
-      compose.services[node_from.normalized_ref].depends_on.push(node_to.normalized_ref);
+      if (!seen_edges.has(`${edge.to}__${edge.from}`)) { // Detect circular refs and pick first one
```

Comment:
This would be another decent candidate for the static string manipulation methods mentioned above.